src/github.com/matrix-org/dendrite/clientapi/auth/storage/accounts/membership_table.go

```diff
@@ -30,6 +30,11 @@ CREATE TABLE IF NOT EXISTS account_memberships (
     room_id TEXT NOT NULL,
     -- The ID of the join membership event
     event_id TEXT NOT NULL,
+	-- Is the user still in the room? We use a boolean instead of just deleting
+	-- the row to know if the user has been in the room at some point
+	-- If set to false, event_id would be the event ID of the leave/kick/ban
+	-- membership event
+	still_in_room BOOLEAN NOT NULL,
```

Comment:
Joined can be misleading imho. I'd interpret it as "the user has joined at some point" without regard to whether he since left or not.